src/cpr_sdk/parser_models.py

```diff
@@ -395,22 +398,31 @@ def to_passage_level_json(self) -> list[dict[str, Any]]:
         if self.text_blocks is None:
             return []
 
-        common_fields_dict = json.loads(
+        fixed_fields_dict = json.loads(
             self.model_dump_json(
                 exclude={
-                    "pdf_data": {"text_blocks", "page_metadata"},
-                    "html_data": {"text_blocks"},
+                    "pdf_data": PDF_DATA_PASSAGE_LEVEL_EXPAND_FIELDS,
+                    "html_data": HTML_DATA_PASSAGE_LEVEL_EXPAND_FIELDS,
                 }
             )
         )
 
         passages_array = [
-            common_fields_dict
+            fixed_fields_dict
             | json.loads(block.model_dump_json(exclude={"text"}))
             | {"text": block.to_string(), "block_index": idx}
             for idx, block in enumerate(self.text_blocks)
         ]
 
+        for passage in passages_array:
+            page_number = passage.get("page_number")
+            if page_number is not None:
+                passage[PDF_PAGE_METADATA_KEY] = self.get_page_metadata_by_page_number(
+                    page_number
+                )
+            else:
+                passage[PDF_PAGE_METADATA_KEY] = None
+
```

Comment:
this might be more readable: `passage[PDF_PAGE_METADATA_KEY] = self.get_page_metadata_by_page_number(page_number) if passage.get("page_number") else None`